src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -412,14 +412,20 @@ static int
 uct_cuda_ipc_md_check_fabric_info(uct_cuda_ipc_md_t *md,
                                   ucs_ternary_auto_value_t mnnvl_enable)
 {
-    int mnnvl_supported = 0;
+    static int mnnvl_supported = -1;
 #if HAVE_NVML_FABRIC_INFO
     nvmlGpuFabricInfoV_t fabric_info;
     nvmlDevice_t device;
     ucs_status_t status;
     char buf[64];
 
-    if (!uct_cuda_base_is_coherent() || (mnnvl_enable == UCS_NO)) {
+    if (mnnvl_supported != -1) {
+        goto out;
+    }
+
+    mnnvl_supported = 0;
```

Comment:
if called from multiple threads, a thread may think mnnvl is not supported while it actually is

Reply:
it should be protected by the lock on UCP level. UCT is not thread-safe